src/Microsoft.AspNetCore.Mvc.Modules/ModularApplicationPart.cs

```diff
@@ -74,18 +72,18 @@ public IEnumerable<TypeInfo> Types
         {
             get
             {
-                return DefaultAssemblyDiscoveryProvider
-                    .DiscoverAssemblies(
-                        ShellBlueprint.Dependencies.Select(dep => dep.Type.GetTypeInfo().Assembly).Distinct().ToList(),
-                        ReferenceAssemblies)
+                return DefaultModularAssemblyDiscoveryProvider
```

Comment:
I seen this because you use the blueprint. So, here it's just a 1st step to show that, in the context of a mvcless app, we only need types for dynamically loaded modules but also their dynamically loaded dependencies.

But i also thought (as before but at the app level) that a tenant could have all modules and then filter enabled ones at another level. So, do we really need this?

If so we can do it. In fact `LoadedAssemblies` are dyn runtimes libraries but for all modules. But extension library can implement a dictionary where each key is related to a module and each value is the list of its dyn runtime assemblies.

In fact, extension library could implement a modular dependency context from which we can retrieve runtime libraries, compilation assemblies ... This for all or each individual module. But 1st maybe better to only do what we need.

So, let me a little time just to filter loaded assemblies, this based on the current blueprint.